base/random/RNGs.jl

```diff
@@ -317,65 +325,63 @@ function rand!(r::MersenneTwister, A::Array{Float64}, n::Int=length(A),
     A
 end
 
+rand!(r::MersenneTwister, A::Array{Float64}, st::StateTrivial{<:FloatInterval_64}) =
+    _rand!(r, A, length(A), st[])
```

Comment:
this is the location of the performance regression? you could try `noinline` here

Reply:
(here or in `_rand!`)

Reply:
Thanks for looking! Actually the performance hit (by about 1x or 2x) doesn't occur here. I will comment where it happens.